web/api/login.go

```diff
@@ -40,23 +37,24 @@ func (dapi *DefaultLoginAPI) LoginPOSTHandler() http.Handler {
 		loginRequest := new(types.RequestLoginPayload).InitFromRequest(r)
 		logrus.Debugf("Login request payload '%v'", loginRequest)
 
-		ok, err := dapi.UserCredentialsDAO.CheckCredentials(loginRequest.Username, loginRequest.Password)
-		if ok {
-			info := dapi.Self.AcceptLoginRequest(
-				loginRequest.Challenge,
-				whisper.AcceptLoginRequestPayload{ACR: "0", Remember: loginRequest.Remember, RememberFor: 3600, Subject: loginRequest.Username},
-			)
-			logrus.Debugf("Accept login request info: %v", info)
-			if info != nil {
-				gohserver.WriteJSONResponse(map[string]interface{}{
-					"redirect_to": info["redirect_to"],
-				}, 200, w)
-				return
-			}
+		userCredential := dapi.UserCredentialsDAO.CheckCredentials(loginRequest.Username, loginRequest.Password)
+
+		if !userCredential.EmailValidated {
+			dapi.Outbox <- mail.GetEmailConfirmationMail(dapi.BaseUIPath, dapi.SecretKey, userCredential.Username, userCredential.Email, loginRequest.Challenge)
```

Comment:
@claudiosegala, here you could get the host header from the request and configure the email link with it.